static/src/js/actions/granules.js

```diff
@@ -170,23 +172,80 @@ export const fetchLinks = retrievalCollectionData => (dispatch, getState) => {
   // Determine how many pages we will need to load to display all granules
   const totalPages = Math.ceil(granuleCount / pageSize)
 
-  const preparedGranuleParams = prepareGranuleAccessParams(granuleParams)
+  const preparedGranuleParams = camelcaseKeys(prepareGranuleAccessParams(granuleParams))
+
+  const graphQuery = `
+    query GetGranuleLinks(
+      $browseOnly: Boolean
+      $circle: [String]
+      $cloudCover: JSON
+      $collectionConceptId: String
+      $conceptId: [String]
+      $dayNightFlag: String
+      $equatorCrossingDate: JSON
+      $equatorCrossingLongitude: JSON
+      $exclude: JSON
+      $limit: Int
+      $line: [String]
+      $linkTypes: [String]
+      $offset: Int
+      $onlineOnly: Boolean
+      $options: JSON
+      $orbitNumber: JSON
+      $point: [String]
+      $polygon: [String]
+      $readableGranuleName: String
+      $sortKey: [String]
+      $temporal: String
+      $twoDCoordinateSystem: JSON
+    ) {
+      granules(
+        browseOnly: $browseOnly
+        circle: $circle
+        cloudCover: $cloudCover
+        collectionConceptId: $collectionConceptId
+        conceptId: $conceptId
+        dayNightFlag: $dayNightFlag
+        equatorCrossingDate: $equatorCrossingDate
+        equatorCrossingLongitude: $equatorCrossingLongitude
+        exclude: $exclude
+        limit: $limit
+        line: $line
+        linkTypes: $linkTypes
+        offset: $offset
+        onlineOnly: $onlineOnly
+        options: $options
+        orbitNumber: $orbitNumber
+        point: $point
+        polygon: $polygon
+        readableGranuleName: $readableGranuleName
+        sortKey: $sortKey
+        temporal: $temporal
+        twoDCoordinateSystem: $twoDCoordinateSystem
+      ) {
+        items{
+          links
+        }
+      }
+    }`
 
   return Promise.all(Array.from(Array(totalPages)).map((_, pageNum) => {
-    const granuleResponse = requestObject.search({
+    const granuleResponse = graphRequestObject.search(graphQuery, {
       ...preparedGranuleParams,
-      pageSize,
-      pageNum: pageNum + 1,
-      echoCollectionId: collectionId
+      limit: pageSize,
+      offset: pageSize * pageNum,
+      linkTypes: ['data', 's3'],
+      collectionConceptId: collectionId
     })
       .then((response) => {
         const { data } = response
-        const { feed } = data
-        const { entry } = feed
+        const { data: granulesData } = data
+        const { granules } = granulesData
+        const { items } = granules
```

Comment:
Remember that this can be null, we should initialize `items` to be an empty array or throw a conditional around the code below this.